app/src/main/cpp/skyline/services/socket/bsd/IClient.cpp

```diff
@@ -1,84 +1,256 @@
 // SPDX-License-Identifier: MPL-2.0
 // Copyright © 2020 Skyline Team and Contributors (https://github.com/skyline-emu/)
 
+#include <poll.h>
 #include "IClient.h"
 
 namespace skyline::service::socket {
     IClient::IClient(const DeviceState &state, ServiceManager &manager) : BaseService(state, manager) {}
 
     Result IClient::RegisterClient(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
-        response.Push<u32>(0);
+        response.Push<i32>(0);
         return {};
     }
 
     Result IClient::StartMonitoring(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
         return {};
     }
 
+    Result IClient::Socket(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        u32 domain{request.Pop<u32>()};
+        u32 type{request.Pop<u32>()};
+        u32 protocol{request.Pop<u32>()};
+        i32 fd{::socket(static_cast<i32>(domain), static_cast<i32>(type), static_cast<i32>(protocol))};
+        Logger::Info("File Descriptor {} with Domain {}, Type {}, Protocol {}", fd, domain, type, protocol);
+        if (fd == -1)
+            Logger::Error("Error creating socket: {}", strerror(errno));
+        return PushBsdResult(response, fd, 0);
+    }
+
     Result IClient::Select(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        response.Push<u32>(0);
```

Comment:
why can this be stubbed?

Reply:
there is some reasons for it: select is pretty annoying to implement, I didn't see any game that uses select because most games prefer use poll instead of select

Reply:
In that case it's best to not implement it to have it print an error

Reply:
You mean logger or errno?

Reply:
He means, remove the implementation of this, and remove the entry from the SFUNC macro, so that an error about unimplemented service is printed to the log.